### ISSUE TITLE
Undefined symbols for architecture armv7

### ISSUE THREAD

**Issue by dwijsoft-dipen** [MailCore/MailCore#143]
When i am try to build and install the application which uses libsasl2.a library it works fine but when i am going to create archive of it to distribute for testing i am getting below errors.

Undefined symbols for architecture armv7:
"_EVP_DigestUpdate", referenced from:
_otp_hash in libsasl2.a(otp.o)
_word2bin in libsasl2.a(otp.o)
"_EVP_DigestFinal", referenced from:
_otp_hash in libsasl2.a(otp.o)
_word2bin in libsasl2.a(otp.o)
"_EVP_get_digestbyname", referenced from:
_generate_otp in libsasl2.a(otp.o)
_otp_server_mech_step in libsasl2.a(otp.o)
"_DES_ede3_cbc_encrypt", referenced from:
_enc_3des in libsasl2.a(digestmd5.o)
_dec_3des in libsasl2.a(digestmd5.o)
"_EVP_cleanup", referenced from:
_otp_common_mech_free in libsasl2.a(otp.o)
"_EVP_DigestInit", referenced from:
_otp_hash in libsasl2.a(otp.o)
_word2bin in libsasl2.a(otp.o)
"_OpenSSL_add_all_digests", referenced from:
_otp_server_plug_init in libsasl2.a(otp.o)
_otp_client_plug_init in libsasl2.a(otp.o)
"_DES_key_sched", referenced from:
_init_des in libsasl2.a(digestmd5.o)
_init_3des in libsasl2.a(digestmd5.o)
"_DES_cbc_encrypt", referenced from:
_enc_des in libsasl2.a(digestmd5.o)
_dec_des in libsasl2.a(digestmd5.o)
ld: symbol(s) not found for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by CodaFi** [comment]
You are linking with libCrypto, right?

**Comment by dwijsoft-dipen** [comment]
Yes i am. Also please note that i am able to install the application on my device through Xcode very well only getting issue when i am trying to create archive of the application to distribute it. 

In my settings, i have tried $(ARCHS_STANDARD_32_BIT) only as well Build Active Architecture Only to NO

**Comment by dwijsoft-dipen** [comment]
I tried move libsasl to under mailcore but getting same error.

**Comment by CodaFi** [comment]
What's in your linker flags?

**Comment by dwijsoft-dipen** [comment]
-ObjC only.

**Comment by dwijsoft-dipen** [comment]
-ObjC only.

**Comment by CodaFi** [comment]
Throw in -lcrypto too.

**Comment by dwijsoft-dipen** [comment]
Cool that's solve my issue. Thanks :)

**Comment by CodaFi** [comment]
Glad I could be of service! :sparkles:

**Comment by dwijsoft-dipen** [comment]
Hey sorry for bugging you again, unfortunately now i am getting error as you can see in the screenshots.
![screen shot 2015-03-08 at 5 56 40 pm](https://cloud.githubusercontent.com/assets/10132614/6545750/dcdf9432-c5bc-11e4-9a8e-b0ec48c5ed3f.png)

Again it is only happening when i am trying to create archive. 

In my target dependancy i have added libsasa2.a and added it my library section. not sure what i am doing wrong.

**Comment by CodaFi** [comment]
When you add a library you also need to let the linker know with -l<name> flags.  Here you want to add `-lsasl2`.
